modules/apps/commerce/commerce-product-asset-categories-web/src/main/resources/META-INF/resources/display_layout/edit_asset_category_cp_display_layout.jsp

```diff
@@ -226,46 +218,44 @@ if (cpDisplayLayout != null) {
 	window.document
 		.querySelector('#<portlet:namespace />selectCategories')
 		.addEventListener('click', (event) => {
-			var itemSelectorDialog = new ItemSelectorDialog.default({
-				eventName: '<portlet:namespace />selectCategory',
-				'strings.add': '<liferay-ui:message key="done" />',
-				title: '<liferay-ui:message key="select-category-display-page" />',
-				url:
-					'<%= categoryCPDisplayLayoutDisplayContext.getCategorySelectorURL(renderResponse) %>',
-			});
-
-			itemSelectorDialog.open();
-
-			itemSelectorDialog.on('selectedItemChange', (event) => {
-				if (event.selectedItem) {
-					var selectedItem = event.selectedItem;
-
-					Object.keys(selectedItem).forEach((key) => {
-						var item = selectedItem[key];
-
-						if (!item.unchecked) {
-							var categoryNodes = categoriesContainer.children;
-
-							for (var i = 0; i < categoryNodes.length; i++) {
-								categoriesContainer.removeChild(
-									categoryNodes.item(i)
+			Liferay.Util.openSelectionModal({
+				buttonAddLabel: '<liferay-ui:message key="done" />',
+				multiple: true,
+				onSelect: (selectedItems) => {
+					if (selectedItems) {
+						Object.keys(selectedItems).forEach((key) => {
+							var item = selectedItems[key];
```

Comment:
Could We use `const/let` on JSPs?

Reply:
I'm never sure about this, some ci:sf's have been complaining about some of the "modern" JS in JSPs. I searched for usages of `const` inside JSPs and there's very little. I also checked a PR or two if they are replacing `var` with `const` but didn't find a satisfying result. @markocikos you probably know, can we replace `var` with `const` inside Commerce JSPs?

Reply:
You can use `const`/`let` in anything that is `master` only — if it is going to be backported to an older branch (where we have to support IE), then you should not.

Reason — we don't transpile JS in JSPs:

- We have SF configured to expect/allow ES6 on `master`, so that means that things like arrow functions, `const`, `let` are all fair game. Newer stuff like nullish coalescing and so on are _not_ ES6, so you can't use them in JSPs.
- On older branches, we expect/allow ES5, so arrows are out, and `const`/`let` too.